ovs-service/src/main/java/org/dcsa/ovs/service/VesselScheduleService.java

```diff
@@ -1,21 +1,172 @@
+        .map(
+            service ->
+                VesselScheduleTO.builder()
+                    .carrierServiceName(service.getCarrierServiceName())
+                    .carrierServiceCode(service.getCarrierServiceCode())
+                    .universalServiceReference(service.getUniversalServiceReference())
+                    .vessels(extractVesselsToVesselTO(service.getVessels()))
+                    .build())
+        .toList();
+  }
+
+  private List<VesselTO> extractVesselsToVesselTO(List<Vessel> vessels) {
+    return vessels.stream()
+        .map(
+            vessel ->
+                VesselTO.builder()
+                    .vesselName(vessel.getName())
+                    .vesselIMONumber(vessel.getImoNumber())
+                    .vesselOperatorCarrierSMDGCode(vessel.getOperatorCarrier().getSmdgCode())
+                    .vesselCallSign(vessel.getCallSign())
+                    .vesselLength(vessel.getLength())
+                    .vesselWidth(vessel.getWidth())
+                    .dimensionUnit(
+                        DimensionUnit.valueFromString(vessel.getDimensionUnit()).orElse(null))
+                    .isDummyVessel(vessel.getIsDummy())
+                    .portCalls(extractTransportCallToPortCallTo(vessel.getPortCalls()))
+                    .build())
+        .toList();
+  }
+
+  private List<PortCallTO> extractTransportCallToPortCallTo(List<TransportCall> transportCalls) {
+    return transportCalls.stream()
+        .map(
+            transportCall ->
+                PortCallTO.builder()
+                    .transportCallReference(transportCall.getReference())
+                    .importVoyageNumber(
+                        Objects.requireNonNullElse(transportCall.getImportVoyage(), new Voyage())
+                            .getCarrierVoyageNumber())
+                    .exportVoyageNumber(
+                        Objects.requireNonNullElse(transportCall.getExportVoyage(), new Voyage())
+                            .getCarrierVoyageNumber())
+                    .importUniversalVoyageReference(
+                        Objects.requireNonNullElse(transportCall.getImportVoyage(), new Voyage())
+                            .getUniversalVoyageReference())
+                    .exportUniversalVoyageReference(
+                        Objects.requireNonNullElse(transportCall.getExportVoyage(), new Voyage())
+                            .getUniversalVoyageReference())
+                    .portTerminalLocation(randomPortTerminalLocation(transportCall))
+                    .portCallStatusCode(
+                        PortCallStatusCode.valueFromString(transportCall.getPortCallStatusCode())
+                            .orElse(null))
+                    .timestamps(extractTransportEventToTimestampTO(transportCall.getTimestamps()))
+                    .build())
+        .toList();
+  }
+
+  private PortTerminalLocation randomPortTerminalLocation(TransportCall transportCall) {
```

Comment:
Is this the correct place to put a Test-generator function (which I assume it is)?